### PR TITLE
add block height check when run transaction

### PR DESCRIPTION
**Pull request by carltraveler**

### CODE REVIEW

File: common/config/config.go [hunk 1]
```diff
@@ -71,8 +71,9 @@ const (
 	DEFAULT_GAS_LIMIT                       = 20000
 	DEFAULT_GAS_PRICE                       = 500
 
-	DEFAULT_DATA_DIR      = "./Chain"
-	DEFAULT_RESERVED_FILE = "./peers.rsv"
+	DEFAULT_DATA_DIR                         = "./Chain"
+	DEFAULT_RESERVED_FILE                    = "./peers.rsv"
+	DEFAULT_OPCODE_HEIGH_UPDATE_FIRST uint32 = 4062597
```
Comment (laizy) [inline common/config/config.go]: need distinguish `testnet`, `mainnet` and `solo` mode, you can reference `STATE_HASH_HEIGHT_MAINNET`

**Comment by Honglei-Cong** [comment]
It's better to set required-engine-version with smart-contract code.
required-engined-version is set when smart-contract is deployed.

When engine decoding smart-contract code, it will compare required-engine-version with op-code engine version.

File: common/constants/constants.go [hunk 2]
```diff
@@ -82,3 +82,7 @@ const (
 // ledger state hash check height
 const STATE_HASH_HEIGHT_MAINNET = 3000000
 const STATE_HASH_HEIGHT_POLARIS = 850000
+
+// neovm opcode update check height
+const OPCODE_HEIGHT_UPDATE_FIRST_MAINNET = 4994530
```
Comment (laizy) [inline common/constants/constants.go]: how this value is calculated?

Comment (carltraveler) [reply common/constants/constants.go]: choose a future block about one month. any better idea?

**Review by laizy — DISMISSED** [review]

**Review by laizy — APPROVED** [review]

**Pull request merged**